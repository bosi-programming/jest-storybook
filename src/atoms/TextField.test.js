import React from "react";
import renderer from "react-test-renderer";
import { TextField } from "./index";

test("TextField", () => {
  const component = renderer.create(<TextField />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("TextField with test label", () => {
  const component = renderer.create(<TextField label="Test label"/>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("TextField with fake data", () => {
  const component = renderer.create(<TextField label="Test label" defaultValue="Test value"/>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
