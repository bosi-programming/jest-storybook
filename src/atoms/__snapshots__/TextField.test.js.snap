// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField 1`] = `
<div
  className="MuiFormControl-root-2"
>
  <div
    className="MuiInputBase-root-19 MuiInput-root-6 MuiInput-underline-10 MuiInputBase-formControl-20 MuiInput-formControl-7"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input-29 MuiInput-input-14"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;

exports[`TextField when addind text 1`] = `
<div
  className="MuiFormControl-root-2"
>
  <label
    className="MuiFormLabel-root-17 MuiInputLabel-root-6 MuiInputLabel-formControl-11 MuiInputLabel-animated-14"
    data-shrink={false}
  >
    Teste label
  </label>
  <div
    className="MuiInputBase-root-37 MuiInput-root-24 MuiInput-underline-28 MuiInputBase-formControl-38 MuiInput-formControl-25"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input-47 MuiInput-input-32"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;

exports[`TextField with fake data 1`] = `
<div
  className="MuiFormControl-root-2"
>
  <label
    className="MuiFormLabel-root-47 MuiInputLabel-root-36 MuiInputLabel-formControl-41 MuiInputLabel-animated-44"
    data-shrink={false}
  >
    Test label
  </label>
  <div
    className="MuiInputBase-root-19 MuiInput-root-6 MuiInput-underline-10 MuiInputBase-formControl-20 MuiInput-formControl-7"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input-29 MuiInput-input-14"
      defaultValue="Test value"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;

exports[`TextField with test label 1`] = `
<div
  className="MuiFormControl-root-2"
>
  <label
    className="MuiFormLabel-root-47 MuiInputLabel-root-36 MuiInputLabel-formControl-41 MuiInputLabel-animated-44"
    data-shrink={false}
  >
    Test label
  </label>
  <div
    className="MuiInputBase-root-19 MuiInput-root-6 MuiInput-underline-10 MuiInputBase-formControl-20 MuiInput-formControl-7"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input-29 MuiInput-input-14"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
</div>
`;
